Ensembl/ensembl-web-track-api
# Add track templates

### Description
Add track payload templates and corresponding script to load tracks for the beta-2 update and beyond.

It has 2 parts:
1) Track templates: each track type has it's own set of values, some of which are internal (e.g. genome browser settings, known by webteam) some external (e.g. track description or datafile name, known by datateams/production). The track templates in this repository is a central place where these fields can be updated and then used to populate Track API.
2) Track loading script: whenever new track datafiles are generated/updated and copied to genome browser NFS area, production can use this script to register corresponding tracks in Track API (to make the tracks available for Ensembl client). The script will read the input datafiles directory and use the corresponding track payload templates to submit the tracks to Track API. The script can be called on specific species/tracks or all available datafiles (see the command-line help text). Web will use this script to load all the tracks for beta-2 update (since the datafiles have already been generated).

### Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2673

### To test

```bash
[ens_adm02@codon-dm-01 ~]$ cd trackapi/utils; export TRACK_API_URL=https://dev-2020.ensembl.org/api/tracks
[ens_adm02@codon-dm-01 utils]$ export TRACK_DATA_DIR=<nfs_path_to_GB_files> # optional in this case
[ens_adm02@codon-dm-01 utils]$ ./submit_track_templates.py -g a7335667-93e7-11ec-a39d-005056b38ce3 -t gc -o
```
This will replace the gc track for pig in dev (you can confirm by comparing the track ID in https://dev-2020.ensembl.org/api/tracks/track_categories/a7335667-93e7-11ec-a39d-005056b38ce3 before and after the script run). Add `-d` to skip track submission and just print the track payload.
